.circleci/infrastructure/commands.yml

```diff
@@ -67,3 +67,49 @@
       - terraform/apply:
           path: <<parameters.tf-path>>
           var_file: ./variables.tfvars
+
+  enable-s3-versioning:
+    parameters:
+      target_env:
+        type: string
+      cf-password:
+        type: env_var_name
+      cf-username:
+        type: env_var_name
+      cf-space:
+        type: string
+    steps:
+      - checkout
+      - run:
+          name: Install dependencies
+          command: |
+            sudo apt update
+            sudo apt install jq
+            sudo apt install curl
+            # TODO: Add Signature check
+            curl -L "https://packages.cloudfoundry.org/stable?release=linux64-binary&version=v7&source=github" | tar -zx
+            sudo mv cf7 /usr/local/bin/cf
+            sudo chmod +x /usr/local/bin/cf
+      - login-cloud-dot-gov:
+          cf-password: <<parameters.cf-password>>
+          cf-username: <<parameters.cf-username>>
+          cf-space: <<parameters.cf-space>>
+      - run:
+          name: Ensure s3 versioning key exists
+          # use cf to check if a service key exists for tdp-datafiles-dev(or prod) 
+          # exit 0 if it does, create it if it doesn't
+          command: |
+            chmod +x ./scripts/create_s3_versioning_key.sh
+            ./scripts/create_s3_versioning_key.sh tdp-datafiles-<<parameters.target_env>> s3-versioning-key
```

Comment:
The logic in this bash script will stop the command here if the key that is used to enable versioning already exists. When the key already exist this indicates that versioning has already been enabled.